lib/crypto/configure.in

```diff
@@ -464,7 +464,6 @@ dnl		so it is - be adoptable
        SSL_APP=
        CRYPTO_APP=
        SSH_APP=
-       AC_MSG_WARN([No (usable) OpenSSL found, skipping ssl, ssh and crypto applications])
```

Comment:
Is this warning only issued when you run --without-ssl ? Because if it is issued also when no openssl installation was found but you wanted it I think it is a relevant warning.

Reply:
I can check that again for you. As far as I know, though, my OpenSSL installation is OK (I've compiled OTP 14.x through 23.x with no issues - changing only OpenSSL versions - and would get that duplicate warning when I disabled it). Let me try and make a matrix of stuff...

Reply:
`./configure --without-ssl`
yields
```
*********************************************************************
**********************  APPLICATIONS DISABLED  **********************
*********************************************************************

crypto         : User gave --without-ssl option
ssh            : User gave --without-ssl option
ssl            : User gave --without-ssl option
ssl            : User gave --without-ssl option
```
(I hadn't tried this option before, I think, for know I see a new duplicate in the output)

---

`./configure --with-ssl=/usr/local/opt/openssl@1.1`
yields no special message (which is what I expect)

---

`./configure --with-ssl=/usr/local/opt/openssl@undefined` (not valid)
yields
```
configure: error: Invalid path to option --with-ssl=PATH
```

Reply:
@IngelaAndin: I think the last test answers your question, but am not totally sure.

Reply:
My question is actually what happens if you run configure without any ssl-flags and there is no OpenSSL installation to be found in any of the usual places? I think that this situation that the warning is aimed for. Both the Erlang ssl and ssh applications are pure Erlang applications that relies on OpenSSL cryptolib linked by the crypto application.

Reply:
@IngelaAndin the row before in the configure output will print

    Checking for OpenSSL >= 0.9.8c in standard locations... no

So, it is the information about the consequence of not finding OpenSSL that will be missing, not that it did not find it.

In my opinion, it does not hurt to be extra clear here, i.e. we should not remove this printout. That way a user can search for `ssh` in the configure log and find this place to see why `ssh` has been disabled.

Reply:
@garazdawi  Yes that was my point that we want to warn for the consequences in this case.

Reply:
> we should not remove this printout

Which one, in particular?

Do you mean reverting the change that removes `AC_MSG_WARN([No (usable) OpenSSL found, skipping ssl, ssh and crypto applications])`?

Reply:
yes.

Reply:
baa4984 handles the duplicate `ssl            : User gave --without-ssl option`.

I'm still checking for the rest.

Reply:
I have this (with simple `.configure` and no with/without options):
```
checking for OpenSSL >= 0.9.8c in standard locations... no
...
*********************************************************************
**********************  APPLICATIONS DISABLED  **********************
*********************************************************************

crypto         : No usable OpenSSL found
ssh            : No usable OpenSSL found
ssl            : No usable OpenSSL found

*********************************************************************
```
(I'd like to confirm this with you, though, since @garazdawi's report seemed to indicate different (when he writes we _should not remove this printout._).

Reply:
(also, if it helps, I'm running `./configure` on the top folder only - not inside any of the sub-folders)

Reply:
You mean that we actually have duplicate warnings about the consequences? That is we get the above without the  AC_MSG_WARN([No (usable) OpenSSL found, skipping ssl, ssh and crypto applications]) and with it we get some additional warning message.

Reply:
Let me try putting the previously removed message there, again, so I can properly reply to your question. I'm getting a bit confused, sorry...

Whatever I'm reporting here is what I've pushed, so I'd ask for your help in verifying if behavior is according to expectation.

Reply:
This is what I get without a locally usable OpenSSL installation (and by reverting the previous change):

```
checking for OpenSSL >= 0.9.8c in standard locations... no
...
configure: WARNING: No (usable) OpenSSL found, skipping ssl, ssh and crypto applications
...
*********************************************************************
**********************  APPLICATIONS DISABLED  **********************
*********************************************************************

crypto         : No usable OpenSSL found
ssh            : No usable OpenSSL found
ssl            : No usable OpenSSL found

*********************************************************************
```

In any case, I'll revert it, no problem. I'm mostly interested in the output after `APPLICATIONS DISABLED` and not the other details, since this is the visible one 😄.

Reply:
I've force-pushed without the previous commit that changed `WARNING: No (usable) OpenSSL found, skipping ssl, ssh and crypto applications`.